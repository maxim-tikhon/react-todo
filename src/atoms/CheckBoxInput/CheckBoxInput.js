import styles from "./CheckBoxInput.module.scss";

const CheckBoxInput = ({ disabled, checked, onCheck }) => {
	return (
		<input
			type="checkbox"
			disabled={disabled}
			checked={checked}
			onChange={e => onCheck(e.target.checked)}
			className={styles.customCheckbox}
		/>
	);
};

export default CheckBoxInput;
