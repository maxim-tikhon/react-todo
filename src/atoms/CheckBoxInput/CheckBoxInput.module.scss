.customCheckbox {
	appearance: none;
	width: 1.5rem;
	height: 1.5rem;
	border-radius: 50%;
	border: 2px solid lightgray;
	background-color: white;
	cursor: pointer;
	position: relative;

	&:not(:disabled):not(:checked):hover {
		border: 2px solid transparent;
		background-image: linear-gradient(white, white),
			radial-gradient(circle at top left, #e983dd80, #579ebfcc);
		background-origin: border-box;
		background-clip: padding-box, border-box;
	}

	&:checked {
		border: 2px solid transparent;
		background-image: linear-gradient(#e983dd80, #579ebfcc),
			radial-gradient(circle at top left, #e983dd80, #579ebfcc);
		background-origin: border-box;
		background-clip: padding-box, border-box;

		&::before {
			opacity: 1;
		}
	}

	&:disabled {
		border: 2px solid lightgrey;
		cursor: not-allowed;
	}

	&::before {
		content: "";
		position: absolute;
		top: 50%;
		left: 50%;
		transform: translate(-50%, -50%);
		width: 12px;
		height: 12px;
		background-image: url("../../../public/images/check.png");
		background-size: contain;
		background-repeat: no-repeat;
		opacity: 0;
		transition: opacity 0.3s ease-in-out;
	}
}
