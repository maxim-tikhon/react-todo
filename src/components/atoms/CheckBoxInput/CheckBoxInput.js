import styles from './CheckBoxInput.module.scss';

function CheckBoxInput({ disabled, checked, label, onCheck }) {
  return (
    <>
      <input
        type="checkbox"
        id={label}
        disabled={disabled}
        checked={checked}
        onChange={(e) => onCheck(e.target.checked)}
        className={styles.customCheckbox}
      />

      {label && (
        <label className={styles.label} htmlFor={label} data-testid="task">
          {label}
        </label>
      )}
    </>
  );
}

export default CheckBoxInput;
