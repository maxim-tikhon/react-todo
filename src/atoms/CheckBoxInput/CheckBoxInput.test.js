import renderer from 'react-test-renderer';
import CheckBoxInput from './CheckBoxInput';

describe('CheckBoxInput', () => {
	it('renders correctly', () => {
		const output = renderer.create(<CheckBoxInput checked={false} />).toJSON();
		expect(output).toMatchSnapshot();
	});
});
