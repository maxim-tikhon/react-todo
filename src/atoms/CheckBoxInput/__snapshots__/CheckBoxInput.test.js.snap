// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBoxInput renders correctly 1`] = `
<input
  checked={false}
  className="customCheckbox"
  onChange={[Function]}
  type="checkbox"
/>
`;
